File: DSA SHEET/DAY-11/RotateAQueue.js
function rotateQueue(queue, k) {
    if(queue.length == 0) return [];
    let res = [];
    while(k-- > 0) {
        queue.push(queue.shift());
    }
    res = queue;
    return res;
}
rotateQueue([10, 20, 30, ], 2);

File: DSA SHEET/DAY-11/RotateAQueue.java
import java.util.*;
public class RotateAQueue {
    public static List<Integer> rotateQueue(Queue<Integer> q, int k) {
        while (k-- > 0) {
            q.add(q.poll());
        }
        List<Integer> ans = new ArrayList<>();
        ans.addAll(new ArrayList<>(q));
        return ans;
    }
    public static void main(String[] args) {
        Queue<Integer> q = new LinkedList<>();
        q.add(10);
        q.add(20);
        q.add(30);
        q.add(40);
        q.add(50);
        System.out.println(rotateQueue(q, 2));
    }
}
